### DISCUSSION TITLE
Auto-Restart on Application Error

### DISCUSSION THREAD

**Discussion by ashermyers** [coollabsio/coolify#1151] [Feature Requests]
Make the application restart when there is an Application Error.

**Comment by hacknug** [comment]
I have a Nuxt 3 site and it keeps going offline on a daily basis. It used to work fine so I'm not sure if it is related to an update (Nuxt or Coolify) or if it is something that I caused myself. Also not sure if this is why it is going offline but it is what I suspect given my limited context (not sure how should I debug this with all these Docker containers involved lol)

**Comment by Flexicon** [comment]
Similarly to the issue requested a while ago, a health-check system would be really nice: https://github.com/coollabsio/coolify/issues/289

**Reply by andrasbacsai** [reply]
V4 has a basic health check system. It will be improved over time to be not just basic.🙂

**Reply by Flexicon** [reply]
Awesome to hear - probably not the best place to ask but is there a relatively simple way to migrate to V4 from V3?

**Reply by andrasbacsai** [reply]
There won't be an automatic upgrade procedure because I cannot make a general function for all the use cases. Maybe I can automate some parts.

Manually, it is easy to migrate your apps and Postgres (these are currently supported in v4).

If you copy the SSL certs (acme.json if you are using Traefik) from the old Coolify server to the new one, deploy your apps on the new one, and change the DNS record to the new server.

If a database is involved, you can make the database public and connect to it from the new server.

Detailed documentation will come soon.
